ui/src/app/base/sagas/http.js

```diff
@@ -0,0 +1,38 @@
+import { call, put, takeLatest } from "redux-saga/effects";
+
+import getCookie from "./utils";
+
+const SCRIPTS_API = `/MAAS/api/2.0/scripts/`;
+
+export const api = {
+  fetchScripts: csrftoken => {
+    const headers = {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      "X-CSRFToken": csrftoken
+    };
+    return fetch(SCRIPTS_API, { headers }).then(response => response.json());
+  }
+};
+
+export function* fetchScriptsSaga() {
+  const csrftoken = yield call(getCookie, "csrftoken");
+  let response;
+  try {
+    yield put({ type: `FETCH_SCRIPTS_START` });
+    response = yield call(api.fetchScripts, csrftoken);
+    yield put({
+      type: `FETCH_SCRIPTS_SUCCESS`,
+      payload: response
+    });
+  } catch (error) {
+    yield put({
+      type: `FETCH_SCRIPTS_ERROR`,
+      error
```

Comment:
I didn't get error in state when I forgot to update `env.development.local` because the error was an instance of Error. Maybe add a check here, `error: error instanceof Error ? error.message : error`... or maybe we should add a card to handle errors more thoroughly.

Reply:
Thanks @Caleb-Ellis, quite right thank you, I've improved the error handling.